ale/base/type_sensor.py

```diff
@@ -466,13 +473,16 @@ def frame_chain(self):
           A networkx frame chain object
         """
         if not hasattr(self, '_frame_chain'):
-            self._frame_chain = FrameChain.from_spice(sensor_frame=self.spacecraft_id * 1000,
+            self._frame_chain = FrameChain.from_spice(sensor_frame=self.final_inst_frame,
                                                       target_frame=self.target_frame_id,
                                                       center_ephemeris_time=self.center_ephemeris_time,
                                                       ephemeris_times=self.ephemeris_time,
                                                       nadir=False, exact_ck_times=False)
             cahvor_quats = Rotation.from_matrix(self.cahvor_rotation_matrix).as_quat()
-            cahvor_rotation = ConstantRotation(cahvor_quats, self.spacecraft_id * 1000, self.sensor_frame_id)
+            #cahvor_rotation = ConstantRotation(cahvor_quats, self.final_inst_frame, self.#sensor_frame_id)
+            cahvor_rotation = ConstantRotation(cahvor_quats, self.target_frame_id,
+                                               self.sensor_frame_id)
```

Comment:
@oleg-alexandrov This is overwriting the frame chain to short circuit from the target to the sensor, I take it as intentional?

Reply:
This is what gives plausible results. After your fix, the orientation changed, but the camera points in the opposite direction of where that hill being imaged is, and there is a very large perspective distortion, and still some shift between the images. Adding my fix on top, the camera now points up, rather than to the left, where the hill is in the DEM, so that's wrong, but there is no shift between successive images and no perspective distortion. So, there is still some missing link in that frame chain.

Reply:
Also note that I removed the temporary hack transform in an earlier commit, so things are more correct now. I also modified one sign in an earlier commit the cahvor_rotation_matrix() function. That was not returning a true rotation, as it had a negative determinant. So, now it really looks if we can get another rotation in the rover frame we should be good.

Reply:
I guess I am more confused now, the `final_inst_frame` was something I added. In doing so I removed the hack rotation that was previous present

Reply:
I see that my independent removal of the hack transform is not even in any history commits, which surely adds to the confusion. I think when I pulled your code I had a conflict with my local commit, and after fixing that things got squashed. Anyhow, I surely agree that your changes made that hack unnecessary so it was right for you to remove it, and we are now incrementally going the right way.

Reply:
👍 That all makes more sense. I am trying to get mapproject working again on my end right now. Running into package versioning issues 😬

Reply:
Fetching ASP with conda in an existing ISIS env is pretty much guaranteed not to work unless your ISIS version is the same which is used with ASP, because every time ISIS gets bumped up packages change and there's so many of them. 

The easiest thing is to fetch and unzip a tarball, as described here: https://stereopipeline.readthedocs.io/en/latest/installation.html#precompiled-binaries-linux-and-macos

Reply:
Will do, running into some weird path binary issues locally with opencv and x11. Worked on friday but an OS update may have caused some issues